src/sentry/tsdb/snuba.py

```diff
@@ -322,6 +275,22 @@ def get_data(
         jitter_value=None,
     ):
         if model in self.non_outcomes_snql_query_settings:
+            # no way around having to explicitly map legacy condition format to SnQL since this function
+            # is used everywhere that expects `conditions` to be legacy format
+            parsed_conditions = []
+            for cond in conditions or ():
+                if not is_condition(cond):
+                    or_conditions = []
+                    for or_cond in cond:
+                        or_conditions.append(parse_condition(or_cond))
+
+                    if len(or_conditions) > 1:
+                        parsed_conditions.append(Or(or_conditions))
+                    else:
+                        parsed_conditions.extend(or_conditions)
+                else:
+                    parsed_conditions.append(parse_condition(cond))
```

Comment:
Taken from:
https://github.com/getsentry/snuba-sdk/blob/ba9185c009ef893d0dd301c2810c1930265d60d6/snuba_sdk/legacy.py#L287-L298

Previously we weren't fully parsing the conditions correctly and left out legacy OR conditions. Not entirely sure about `AND` conditions though.

Reply:
Actually, not entirely sure the absence of this bit of code was the cause of the list index out of range issue.

Reply:
Does your test fail if you remove this change?

Reply:
Yeah, I made my original comment before adding the test. Commenting out the `if not is_condition(cond)` branch does reproduce the exception referenced in the issue.